### PR TITLE
Feature/fuse maybe

### PR DESCRIPTION
**Pull request by smoser**
Lots of debug here.
definitely work in progress. locally it passes a few small tests for me.

### CODE REVIEW

**Comment by rchincha** [comment]
Would squash it once done.

**Comment by smoser** [comment]
> Would squash it once done.

I think it is done. I'd prefer to leave it as 4 commits. it is 4 units of work.

**Review by hallyn — APPROVED** [review]

File: cmd/lxc-wrapper/lxc-wrapper.c [hunk 1]
```diff
@@ -71,102 +80,153 @@ static int map_id(char *args[])
...
+	// The intent is that our child process is in:
+	//  * user namespace for 'usernsexec' (CLONE_NEWUSER)
+	//  * private (MS_PRIVATE) mount namespace always (CLONE_NEWNS)
+	//  * pid namespace always (CLONE_NEWPID)
+	//
+	// If not in its' own user namespace, then mounts will propagate.
+	// change that behavior so it is consistent for both cases.
+	ret = mount("none", "/", 0, MS_PRIVATE | MS_REC, NULL);
+	if (ret != 0) {
+		fprintf(stderr, "mounting proc failed: %m\n");
```
Comment (smoser) [inline cmd/lxc-wrapper/lxc-wrapper.c:99]: Ugh. Mounting proc failed. Should change that message.

**Comment by smoser** [comment]
I guess one thing that I'd really like to have added.... but I'm not wanting to do it right now, is a test case like shown in #319.

These changes fix that test case (as I commented in the issue).

Basically, before this commit, we could not do "squashfs only" builds either privileged or unprivileged without starting
`from` a `type: docerk` `url` that referenced tar content.

**Pull request merged**